### PR TITLE
[core] Support setting the time field type for record level expire

### LINKED ISSUE #3989: [Feature] Support setting the filed type of record-level.time-field
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.

### Motivation

Currently the time field for record level expire must be a seconds INT, however，the timestamps involved in business practice are often millisecond timestamps. So I want to support setting the time field type for record level expire, it can be second or millisecond.

### Solution

I will use the record-level.time-field.type to set the time field type for record level expire, it can be second or millisecond. You can review the commit in this PR https://github.com/chenxinwei/paimon/pull/1/commits/ed951a4e1782039e52fd0a5d557dafd9990d1995. 

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by chenxinwei**
Change-Id: Iced3e7a7e6facb3cd37c87a2ee87723c9ba6d5ab

### Purpose

Linked issue: close #3989 

Support setting the time field type for record level expire, it can be second or millisecond.

### Tests
Add a UT int RecordLevelExpireWithMillisecondTest.java

### CODE REVIEW

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 1]
```diff
@@ -1251,8 +1251,14 @@ public class CoreOptions implements Serializable {
             key("record-level.time-field")
                     .stringType()
                     .noDefaultValue()
+                    .withDescription("Time field for record level expire.");
+
+    public static final ConfigOption<TimeFieldType> RECORD_LEVEL_TIME_FIELD_TYPE =
+            key("record-level.time-field.type")
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:1257]: record-level.field-type

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 2]
```diff
@@ -2682,4 +2693,33 @@ public InlineElement getDescription() {
             return text(description);
         }
     }
+
+    /** Time field type for record level expire. */
+    public enum TimeFieldType implements DescribedEnum {
+        SECOND(
+                "second",
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:2700]: `seconds-int`

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 3]
```diff
@@ -2682,4 +2693,33 @@ public InlineElement getDescription() {
...
+                "10-bit timestamp which indicates the number of seconds from 00:00:00 UTC on January 1, 1970 to the present, excluding milliseconds."),
+
+        MILLISECOND(
+                "millisecond",
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:2704]: `millis-long`

File: paimon-core/src/main/java/org/apache/paimon/io/RecordLevelExpire.java [hunk 4]
```diff
@@ -85,7 +92,10 @@ private RecordReader<KeyValue> wrap(RecordReader<KeyValue> reader) {
                     checkArgument(
                             !kv.value().isNullAt(timeField),
                             "Time field for record-level expire should not be null.");
-                    int recordTime = kv.value().getInt(timeField);
+                    int recordTime =
+                            timeFieldType == CoreOptions.TimeFieldType.SECOND
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/io/RecordLevelExpire.java:96]: Use switch case.

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java (hunk 1 above)
Comment (JingsongLi) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:1257]: It is OK

Comment (chenxinwei) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java]: Fixed in https://github.com/apache/paimon/pull/3992/commits/20a186ed4c7abbf08f9b89ae33f74a5075d068dd.

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java (hunk 2 above)
Comment (chenxinwei) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java]: Fixed in https://github.com/apache/paimon/commit/20a186ed4c7abbf08f9b89ae33f74a5075d068dd

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java (hunk 3 above)
Comment (chenxinwei) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java]: Fixed in https://github.com/apache/paimon/commit/20a186ed4c7abbf08f9b89ae33f74a5075d068dd

File: paimon-core/src/main/java/org/apache/paimon/io/RecordLevelExpire.java (hunk 4 above)
Comment (chenxinwei) [reply paimon-core/src/main/java/org/apache/paimon/io/RecordLevelExpire.java]: Fixed in https://github.com/apache/paimon/commit/20a186ed4c7abbf08f9b89ae33f74a5075d068dd

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**